components/shared/fuzzy-match.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react'
+import useTheme from '../styles/use-theme'
+import withDefaults from '../utils/with-defaults'
+import { tuple } from '../utils/prop-types'
+
+const patterns = tuple('fuzzy', 'exact', 'regex')
+
+type Patterns = typeof patterns[number]
+
+const colorTypes = tuple('secondary', 'primary')
+
+type Color = typeof colorTypes[number]
+
+interface Props {
+  color?: string
+  label: string
+  query: string
+  pattern?: Patterns
+  height: string
+}
+
+const defaultProps = {
+  pattern: 'fuzzy' as Patterns,
+  color: 'default' as Color,
+}
+
+type NativeAttrs = Omit<React.HTMLAttributes<any>, keyof Props>
+export type FuzzyMatchProps = Props & typeof defaultProps & NativeAttrs
+
+const FuzzyMatchComponent: React.FC<React.PropsWithChildren<FuzzyMatchProps>> = ({
+  color: colorType,
+  className,
+  label,
+  query,
+  pattern,
+  height,
+  ...props
+}) => {
+  const theme = useTheme()
+  const color = { secondary: theme.palette.cNeutral5, primary: 'inherit' }[colorType]
+  const highlightColor = { secondary: theme.palette.cNeutral7, primary: theme.palette.cTheme5 }[
+    colorType
+  ]
+
+  if (pattern === 'fuzzy') {
+    query = query.trim().replace(/\ /, '.*')
+  }
+
+  let beforeMatched = ''
+  let matched = ''
+  let afterMatched = ''
+  if (pattern === 'fuzzy' || pattern === 'regex') {
```

Comment:
bug: set pattern to `regex`, eg: `<AutoComplete placeholder="Enter here" options={options} pattern="regex" />` 

then input `?`, throw `SyntaxError: Invalid regular expression: /?/: Nothing to repeat`

Reply:
the same goes for direct input, try `?`, `[` ...

Reply:
Done. I removed the `pattern` prop.